render/render.js

```diff
@@ -167,24 +173,20 @@ module.exports = function($window) {
 		else if (vnodes == null) removeNodes(old, 0, old.length, vnodes)
 		else {
 			if (old.length === vnodes.length) {
-				var isUnkeyed = false
 				for (var i = 0; i < vnodes.length; i++) {
-					if (vnodes[i] != null && old[i] != null) {
-						isUnkeyed = vnodes[i].key == null && old[i].key == null
-						break
-					}
-				}
-				if (isUnkeyed) {
-					for (var i = 0; i < old.length; i++) {
-						if (old[i] === vnodes[i]) continue
-						else if (old[i] == null && vnodes[i] != null) createNode(parent, vnodes[i], hooks, ns, getNextSibling(old, i + 1, nextSibling))
-						else if (vnodes[i] == null) removeNodes(old, i, i + 1, vnodes)
-						else updateNode(parent, old[i], vnodes[i], hooks, getNextSibling(old, i + 1, nextSibling), recycling, ns)
+					if (vnodes[i] != null && old[i] != null &&
+							vnodes[i].key == null && old[i].key == null) {
```

Comment:
This is probably not a win...

The old `unkeyed` detection loop stops as soon as two non-null nodes are found.

The new code will iterate until the end of keyed lists.

Keyed diff isn't covered by the perf code, which may explain why you missed this.